### ISSUE TITLE
q is not a function

### ISSUE THREAD

**Issue by mgreschke** [LukaszWatroba/v-modal#4]
Hello!

First of all thanks for this awesome modal.. definitely amazing UI help :)

I ran into trouble with my code getting the following error (im using Angular 1.3.5):

Uncaught TypeError: q is not a function(anonymous function) @ angular-animate.js:936 on page load and modal not showing on onClick. vModal is the only module using ngAnimate.

the relevant code:

```
var marketPlaceApp = angular.module('marketPlace', ['user', 'productService', 'ngAnimate', 'vModal']);

        marketPlaceApp.config(function ($compileProvider) {
            $compileProvider.debugInfoEnabled(false);
        });

       marketPlaceApp.factory('addToCartModal', function (vModal) {
           return vModal({
               controller: 'addToCartController',
               controllerAs: 'addToCartModal',
               templateUrl: 'cart-modal-template.html'
           });
       });

       marketPlaceApp.factory('ocpModal', function (vModal) {
           return vModal({
               controller: 'ocpController',
               controllerAs: 'ocpModal',
               templateUrl: 'ocp-modal-template.html'
           });
       });

       marketPlaceApp.controller('addToCartController', function ($scope, addToCartModal) {
           var ctrl = this;
           ctrl.close = addToCartModal.deactivate;
       });


       marketPlaceApp.controller('ocpController', function ($scope, ocpModal) {
           var ctrl = this;
           ctrl.close = ocpModal.deactivate;
       });

call in main controller function:

 vm.addToCart = function addToCart (product){
                 $log.log('activating cart modal');
                 vm.openAddToCartModal = addToCartModal.activate
             };
```

Any idea how to fix this?

cheers

**Comment by LukaszWatroba** [comment]
Could you post a JS Fiddle demonstrating the problem?

**Comment by mgreschke** [comment]
Issue was on my side, mistakenly included 1.2.x version of angular-animate which didnt support promises.

thanks anyways!
